File: challenges/char-count/js/charCount.js
const charCount = (str) => {
    const counts = {} // creates empty object
    const letters = str.split('').filter((char) => { // splits string into list of characters
        return char !== ' '
    })
    for (let char of letters) { // adds characters to object with count
        counts[char] = isNaN(counts[char]) ? 1 : counts[char] += 1
    }
    return counts // returns the final object
};

module.exports = { charCount }

File: challenges/char-count/python/char_count.py
from collections import defaultdict


def char_count(str):
    counts = defaultdict(int)  # default is an int
    letters = [char for char in str if char != ' ']  # list of characters
    for char in letters:
        counts[char] += 1  # one is default value
    return counts  # returns final dict object
